client/go/outline/electron/vpn.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"log/slog"
+	"sync"
+
+	perrs "github.com/Jigsaw-Code/outline-apps/client/go/outline/platerrors"
+)
+
+type vpnConfigJSON struct {
+	ID              string   `json:"id"`
+	InterfaceName   string   `json:"interfaceName"`
+	IPAddress       string   `json:"ipAddress"`
+	DNSServers      []string `json:"dnsServers"`
+	RoutingTableId  int      `json:"routingTableId"`
+	RoutingPriority int      `json:"routingPriority"`
+	ProtectionMark  uint32   `json:"protectionMark"`
+	TransportConfig string   `json:"transport"`
+}
+
+type vpnConnectionJSON struct {
+	ID       string `json:"id"`
+	Status   string `json:"status"`
+	RouteUDP *bool  `json:"routeUDP"`
+}
+
+type VPNStatus string
+
+const (
+	VPNConnected     VPNStatus = "Connected"
```

Comment:
Since this is already in the vpn package, you are stuttering: `vpn.VPNConnected`.
Perhaps better to have `vpn.StatusConnected`

See https://go.dev/doc/effective_go#package-names

![image](https://github.com/user-attachments/assets/6ad5e9b6-70db-4e97-aa6c-753c47fa5883)

Reply:
Sampe for the `VPN*` types

Reply:
But this is not in `vpn` package, it's electron's `main` package. How about moving all these to a dedicated `vpn` package, and then merging `vpnlinux` with it? Also I can move the Method Call logic to `outline.InvokeMethod`

Reply:
Done.